### ISSUE TITLE
should override os.path.normcase on Windows

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jmcgeheeiv/pyfakefs#28]
```
FakeFilesystem takes a path_separator arg, primarily useful for cross-platform 
faking on Windows.  However it doesn't override os.path.normcase, which on 
Windows replaces forward-slashes with backslashes.

What steps will reproduce the problem?
import fake_filesystem
filesystem = fake_filesystem.FakeFilesystem(path_separator='/')
fake_os = fake_filesystem.FakeOsModule(filesystem)
os = fake_os                    # bash real os
print os.path.normcase('/tmp')

What is the expected output?
/tmp

What do you see instead?
\tmp

What version of the product are you using? On what operating system?
2.4, on Windows

This causes problems with e.g. the tempfile module, because it uses normcase 
which produces invalid filenames in this situation.

I think the simplest solution is to just override os.path.normcase with a null 
implementation that returns its arg on Windows, if the filesystem's 
path_separator is '/'.
```

Original issue reported on code.google.com by `ga...@oberbrunner.com` on 2 May 2014 at 3:41

**Comment by jmcgeheeiv** [comment]
I do not have a Windows development environment set up.  It would be great if someone who does would take care of this (seemingly) simple bug.

**Comment by mrbean-bremen** [comment]
The problem is that under Windows there are actually 2 more or less valid path separators (\ and /), and using a fake file system with one of these will be a problem if handling paths that contain the other one. 
Avoiding this is possible but sometimes cumbersome if the paths come from configuration files etc. and you don't want to adapt the production code all over the place, so I fixed the problem for myself by patching some of the FakeFilesystem methods (CollapsePath and some more) by a dumb replace('/', os.sep) on the argument (I used '\' as separator, obviously).
Not sure if this is a good idea (it is somewhat ugly), but it does the job.
